src/test_typing.py

```diff
@@ -893,6 +893,12 @@ class MyDef(typing.DefaultDict[str, T]): ...
         self.assertIs(MyDef[int]().__class__, MyDef)
         self.assertIs(MyDef[int]().__orig_class__, MyDef[int])
 
+    @skipUnless(sys.version_info >= (3, 3), 'ChainMap was added in 3.3')
+    def test_chainmap_type_erasure_special(self):
+        class MyChain(typing.ChainMap[str, T]): ...
+        self.assertIs(MyChain[int]().__class__, MyChain)
+        self.assertIs(MyChain[int]().__orig_class__, MyChain[int])
```

Comment:
One last thing, you use ``assertIs`` here that relies on generic type cache. This is a bit dangerous if the test is run in refleak hunting mode (-R). I see two options here:
* change this to use ``assertEqual``
* add ``self.clear_caches()`` at beginning and edit the comment in the previous ``test_type_erasure_special`` to say something like ``# this test and the next one are only two that rely on type caching``

Sorry for not noticing this before.